app/web/src/store/views.store.ts

```diff
@@ -1846,6 +1847,7 @@ export const useViewsStore = (forceChangeSetId?: ChangeSetId) => {
                 // then refetch (i.e. there might be updates!)
                 if (data.toRebaseChangeSetId === changeSetId) {
                   this.FETCH_VIEW();
+                  componentsStore.FETCH_ALL_COMPONENTS();
                   this.LIST_VIEWS();
```

Comment:
It feels like this could cause the same race that was this PR fixes, so it makes sense to make this func async and await these calls too

Reply:
Yeah I had a similar thought too - good shout.